src/shared/components/axisTicks/TimeTickInput.tsx

```diff
@@ -0,0 +1,236 @@
+  } = props
+
+  const [tickOptionInput, setTickOptionInput] = useOneWayState(
+    initialTickOptionValue === initialTickOptionValue
+      ? initialTickOptionValue
+      : ''
+  )
+
+  const [tickOptionInputStatus, setTickOptionInputStatus] = useState<
+    ComponentStatus
+  >(ComponentStatus.Default)
+
+  const [isDatePickerOpen, setIsDatePickerOpen] = useState<boolean>(false)
+
+  const [
+    isOnClickOutsideHandlerActive,
+    setIsOnClickOutsideHandlerActive,
+  ] = useState<boolean>(true)
+
+  const triggerRef: RefObject<ButtonRef> = createRef()
+
+  const updateTickOption = (value?: string | number) => {
+    const convertedValue = convertUserInputValueToNumOrNaN(
+      value === undefined ? tickOptionInput : value
+    )
+    const tickOptionNameWithoutAxis = label.split(' ').join('')
+    const tickOptionNameWithAxis = `${axisName.toLowerCase()}${tickOptionNameWithoutAxis}`
+    const filteredTickOptions = Array.isArray(tickOptions)
+      ? tickOptions.filter(option => option !== tickOptionNameWithAxis)
+      : []
+    setOptions(tickOptionNameWithoutAxis, convertedValue)
+    if (convertedValue === convertedValue) {
+      setOptions('GenerateAxisTicks', [
+        ...filteredTickOptions,
+        tickOptionNameWithAxis,
+      ])
+    } else {
+      setOptions('GenerateAxisTicks', filteredTickOptions)
+      if (tickOptionInput !== '') {
+        setTickOptionInputStatus(ComponentStatus.Error)
+      }
+    }
+  }
+
+  const handleInput = (event: ChangeEvent<HTMLInputElement>) => {
+    setTickOptionInput(event.target.value)
+    setTickOptionInputStatus(ComponentStatus.Default)
+  }
+
+  const handleBlur = () => updateTickOption()
+
+  const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      updateTickOption()
+    }
+  }
+
+  const getDatePickerDateTime = () => {
+    const date = new Date(tickOptionInput)
+    if (date.valueOf() === date.valueOf()) {
+      return date.toISOString()
+    }
+    return new Date().toISOString()
+  }
+
+  const handleReset = () => {
+    setTickOptionInput('')
+    updateTickOption('')
+  }
+
+  const showDatePicker = () => setIsDatePickerOpen(true)
+  const hideDatePicker = () => setIsDatePickerOpen(false)
+
+  const toggleDatePicker = () => {
+    if (isDatePickerOpen) {
+      hideDatePicker()
+    } else {
+      showDatePicker()
+    }
+  }
+
+  const handleSelectDate = (date: string) => {
+    const dateRFC3339 = new Date(date).valueOf()
+    setTickOptionInput(dateRFC3339 === dateRFC3339 ? String(dateRFC3339) : '')
+    setTickOptionInputStatus(ComponentStatus.Default)
+    updateTickOption(dateRFC3339)
+  }
+
+  const onClickOutside = () => {
+    if (isOnClickOutsideHandlerActive) {
+      hideDatePicker()
+    }
+  }
+
+  const allowOnClickOutside = () => setIsOnClickOutsideHandlerActive(true)
+  const suppressOnClickOutside = () => setIsOnClickOutsideHandlerActive(false)
+
+  const style: CSSProperties = isDatePickerOpen
```

Comment:
i think conventionally it's called `styles`

Reply:
`styles` it is. Done.